### PR TITLE
feat: multisig_v2

### PR DESCRIPTION
**Pull request by WasinWatt**
# Description

Closes: #XXXX

## Notes on how this MultiSig wallet works

1. Able to execute newer proposals even if the older ones are still active like existing `0x1::multisig`
2. Support multisig wallet with tiers and weights
3. Changing threshold and membership affects active proposals accordingly, this means
    1. casted votes which belong to the voters that got removed by `update_config` proposal will be removed from active proposals, meaning voters info of active proposals can be changed by `update_config`
    2. threshold of active proposals also changes accordingly
    4. this is completely different from exisitng `0x1::multisig` where current active proposals will be voided if config changes
    5. this doesn't affect executed and expired proposals
4. Expiry timestamp will be in Proposal struct instead of Multisig meaning different proposal can have different expiry. Our frontend application will have the default of 7 days.
5. Only support expiry timestamp. Height expiry is removed.

Figma reference: [figma - multisig](https://www.figma.com/design/iMu1XHfOWDMzqWLHS1aAuX/MultiSig?node-id=181-50787&t=8HcT5vZcOShMQh3o-1)

P.S. this is my first time writing `move`, feel free to comment

---

## Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:
  
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] included the necessary unit and integration tests
- [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] confirmed all CI checks have passed

## Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

## Summary by CodeRabbit

- **New Features**
	- Introduced a multisignature wallet system for shared wallet management.
	- Added functionality for creating weighted and non-weighted multisig accounts.
	- Enabled proposal creation and voting mechanisms for executing blockchain functions.
	- Implemented configuration updates for multisig wallets, including member and tier management.
	- Added view functions for retrieving multisig account and proposal information.
- **Bug Fixes**
	- Enhanced error handling with defined constants for error codes and statuses.
- **Tests**
	- Comprehensive test cases to validate multisig wallet functionality across various scenarios.

### CODE REVIEW

**Review by beer-1 — COMMENTED** [review]
LGTM; left minor correction in style.

**Review by beer-1 — APPROVED** [review]
LGTM

File: precompile/modules/initia_stdlib/sources/multisig_v2.move [hunk 1]
```diff
@@ -162,6 +159,13 @@ module initia_std::multisig_v2 {
         tiers: Option<vector<Tier>>
     }
 
+    // view functions
+    public fun is_exist(creator_addr: address, name: &String): bool {
```
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/multisig_v2.move:163]: If it is view functions, then we need to put `#[view]`

**Review by beer-1 — APPROVED** [review]
LGTM but need to add 1 missing `#[view]` attribute.

File: precompile/modules/initia_stdlib/sources/multisig_v2.move (hunk 1 above)
Comment (WasinWatt) [reply precompile/modules/initia_stdlib/sources/multisig_v2.move:165]: oops, done

**Pull request merged**